### PR TITLE
wifi-subsys: Adding sntp support to sync time

### PR DESCRIPTION
**Pull request by eduazocar**
### Contribution description

Integrations of sntp protocol to get  the current date, time from an zone in specific. 

this implement two functions, init_sntp to connect with an sntp server anf get_time, that works setting the time of the system

### Issues/PRs references

Fixes #134

### CODE REVIEW

File: wifi-subsys/components/network/src/sntp.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/* LwIP SNTP example
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
```
Comment (xkevin190) [inline wifi-subsys/components/network/src/sntp.c:8]: remove this

File: wifi-subsys/components/network/src/sntp.c [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#include "esp_sleep.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/event_groups.h"
+#include "freertos/task.h"
+#include "nvs_flash.h"
+#include <string.h>
+#include <sys/time.h>
+#include <time.h>
```
Comment (xkevin190) [inline wifi-subsys/components/network/src/sntp.c:21]:
```suggestion
#include <string.h>
#include <sys/time.h>
#include <time.h>
#include "esp_attr.h"
#include "esp_event_loop.h"
#include "esp_log.h"
#include "esp_sleep.h"
#include "esp_system.h"
#include "esp_wifi.h"
#include "freertos/FreeRTOS.h"
#include "freertos/event_groups.h"
#include "freertos/task.h"
#include "nvs_flash.h"

```

File: wifi-subsys/components/network/src/sntp.c [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+/* The examples use simple WiFi configuration that you can set via
+   'make menuconfig'.
+
+   If you'd rather not, just change the below entries to strings with
+   the config you want - ie #define EXAMPLE_WIFI_SSID "mywifissid"
+*/
+
+/* FreeRTOS event group to signal when we are connected & ready to make a
+ * request */
```
Comment (xkevin190) [inline wifi-subsys/components/network/src/sntp.c:33]: remove this.

**Review by xkevin190 — CHANGES_REQUESTED** [review]

**Comment by xkevin190** [comment]
where is .h file?

File: wifi-subsys/components/network/src/sntp.c (hunk 1 above)
Comment (luisan00) [reply wifi-subsys/components/network/src/sntp.c:8]: Well, if this job is based on that job the original header license should stay, only  add our copyright

File: wifi-subsys/components/network/src/protocol_sntp.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * @brief this function sync with a server sntp to get a current time and date, this format is
+ * returned in milliseconds (time posix)
+ * @param [inout] curr_time* This is a pointer param to manage the time, this take a time_t var and
+ * manage if was set in the system.
+ * @param [inout] curr_timeinfo* pointer structure to detailed info of the time_t var, in this case
+ * curr_time*
+ * @param [in] time_format This is en input to specific the time format to use UTC. GMT...
+ *
+ */
+
```
Comment (luisan00) [inline wifi-subsys/components/network/src/protocol_sntp.h:54]:
```suggestion
```

File: wifi-subsys/components/network/src/protocol_sntp.h [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ *
+ * @brief       sntp (Simple Network Time Protocol)
+ * @ingroup     wifi-subsys_components
+ * @{
+ * @author      eduazocar <eduardo@turpialdev.com>
+ *
+ */
```
Comment (luisan00) [inline wifi-subsys/components/network/src/protocol_sntp.h:24]:
```suggestion
/**
 * @ingroup     wifi-subsys_components
 * @{
 * @brief       sntp (Simple Network Time Protocol)
 * @author      eduazocar <eduardo@turpialdev.com>
 */
```

File: wifi-subsys/components/network/test/test_wifi.c [hunk 6]
```diff
@@ -95,4 +95,4 @@ TEST_CASE("change auth ap", "[network]")
     if(err != ESP_OK) {
         TEST_FAIL();
     }
-}
+}
```
Comment (luisan00) [inline wifi-subsys/components/network/test/test_wifi.c:98]: new line at  EOF

**Review by luisan00 — CHANGES_REQUESTED** [review]
take  another look

**Review by luisan00 — APPROVED** [review]
let's go ahead! :fire:

**Review by xkevin190 — APPROVED** [review]

**Pull request merged**